roles/jd-server/src/lib/job_declarator/mod.rs

```diff
@@ -1,27 +1,30 @@
 use crate::{error::PoolError, Configuration, EitherFrame, StdFrame};
 use async_channel::{Receiver, Sender};
-use binary_sv2::B0255;
+use binary_sv2::{Seq0255, B0255, U256};
 use bitcoin::consensus::Encodable;
 use codec_sv2::{Frame, HandshakeRole, Responder};
+use ed25519_dalek::{Keypair, PublicKey, Signature, SignatureError, Signer, Verifier};
 use error_handling::handle_result;
 use network_helpers::noise_connection_tokio::Connection;
+use noise_sv2::formats::{EncodedEd25519PublicKey, EncodedEd25519SecretKey};
 use roles_logic_sv2::{
     common_messages_sv2::SetupConnectionSuccess,
     handlers::job_declaration::{ParseClientJobDeclarationMessages, SendTo},
     job_declaration_sv2::{AllocateMiningJobTokenSuccess, CommitMiningJobSuccess, *},
     parsers::{JobDeclaration, PoolMessages},
     utils::Mutex,
 };
-use std::{convert::TryInto, sync::Arc};
+use std::{convert::TryInto, str, sync::Arc};
 use tokio::net::TcpListener;
-use tracing::info;
+use tracing::{debug, info};
 
 #[derive(Debug)]
 pub struct JobDeclaratorDownstream {
     sender: Sender<EitherFrame>,
     receiver: Receiver<EitherFrame>,
     // TODO this should be computed for each new template so that fees are included
     coinbase_output: Vec<u8>,
+    config: Configuration,
```

Comment:
We do not need to have all the config data here. Also best practice is to not derive Debug on struct that contains sensible data (like the pool private key). You can either implement Debug manually and avoid to log the secret or using something like https://docs.rs/secrecy/latest/secrecy/struct.Secret.html

Reply:
I'll give a look to the struct.Secret and implement this solution